skypeweb/skypeweb_contacts.c

```diff
@@ -924,35 +924,23 @@ skypeweb_received_contacts(SkypeWebAccount *sa, PurpleXmlNode *contacts)
 	purple_notify_searchresults(sa->pc, _("Received contacts"), NULL, NULL, results, NULL, NULL);
 }
 
-void
-skypeweb_search_users_text_cb(SkypeWebAccount *sa, JsonNode *node, gpointer user_data)
+static PurpleNotifySearchResults*
+create_search_results(JsonNode *node, gint *olength)
 {
+	PurpleNotifySearchColumn *column;
+	gint index, length;
 	JsonObject *response = NULL;
 	JsonArray *resultsarray = NULL;
-	gint index, length;
-	gchar *search_term = user_data;
-
-	PurpleNotifySearchResults *results;
-	PurpleNotifySearchColumn *column;
 	
 	response = json_node_get_object(node);
 	resultsarray = json_object_get_array_member(response, "results");
 	length = json_array_get_length(resultsarray);
 	
-	if (length == 0)
+	PurpleNotifySearchResults *results = purple_notify_searchresults_new();
+	if (results == NULL || length == 0)
 	{
-		gchar *primary_text = g_strdup_printf("Your search for the user \"%s\" returned no results", search_term);
-		purple_notify_warning(sa->pc, "No users found", primary_text, "", purple_request_cpar_from_connection(sa->pc));
-		g_free(primary_text);
-		g_free(search_term);
-		return;
-	}
-	
-	results = purple_notify_searchresults_new();
-	if (results == NULL)
-	{
-		g_free(search_term);
-		return;
+		*olength = 0;

```

Comment:
Needs some null-check safety

Reply:
Would you be okay with an assert? This is really a function that's meant to be used internally with `*olength` pointing to a local variable, as you can see in the current two usages.

Reply:
asserts are evil :)

Adding more safety (even for an internal function) is always useful for future use.

Reply:
As you wish :)
Changes pushed.